pkg/metrics/metrics.go

```diff
@@ -0,0 +1,44 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	runtimemetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+// When adding metric names, see https://prometheus.io/docs/practices/naming/#metric-names
+const (
+	existingKMMOCRsQuery = "kmmo_number_crs_total"
+)
+
+//go:generate mockgen -source=metrics.go -package=metrics -destination=mock_metrics_api.go
+
+// Metrics is an interface representing a prometheus client for the Special Resource Operator
+type Metrics interface {
+	Register()
+	SetExistingKMMOsCRs(value int)
+}
+
+func New() Metrics {
+
+	kmmoResourcesNum := prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: existingKMMOCRsQuery,
+			Help: "Number of existing KMMO CRs",
+		},
+	)
+	return &metrics{kmmoResourcesNum: kmmoResourcesNum}
+}
+
+func (m *metrics) Register() {
+	runtimemetrics.Registry.MustRegister(
+		m.kmmoResourcesNum,
+	)
+}
+
+type metrics struct {
```

Comment:
Can we move that declaration just beneath `Metrics` interface just for convenience?